src/Chapter3.hs

```diff
@@ -907,9 +1023,34 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+newtype Gold = Gold { goldValue :: Int }
 class Append a where
-    append :: a -> a -> a
+	append :: a -> a -> a
+
+instance Append Gold where
+	append :: Gold -> Gold -> Gold
+	append x1 x2 = Gold ((goldValue x1) + (goldValue x2))
 
+instance Append [a] where
+	append :: [a] -> [a] -> [a]
+	append x1 x2 = x1 ++ x2
+
+instance (Append a) => Append (Maybe a) where
+	append :: Maybe a -> Maybe a -> Maybe a
+	append (Just x1) (Just x2) = Just (append x1 x2)
+	append _ _ = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality it would be more beneficial is this function returned `Just` when at least one argument is `Just`.